src/test/kotlin/com/ampnet/auditornode/controller/websocket/AuditWebSocketUnitTest.kt

```diff
@@ -0,0 +1,810 @@
+                .sendSync(connectedMessage)
+            then(session)
+                .should(times(1))
+                .sendSync(responseMessage)
+        }
+
+        verify("web socket session was closed") {
+            then(session)
+                .should(times(1))
+                .close(CloseReason.ABNORMAL_CLOSURE)
+        }
+    }
+
+    @Test
+    fun `must return error when auditing procedure script file cannot be fetched`() {
+        val sessionAttributes = mock<MutableConvertibleValues<Any>>()
+        val session = mock<WebSocketSession> {
+            on { attributes } doReturn sessionAttributes
+        }
+
+        val connectedMessage = "connected"
+        val responseMessage = "error"
+
+        suppose("web socket messages will be serialized") {
+            given(objectMapper.writeValueAsString(ConnectedInfoMessage))
+                .willReturn(connectedMessage)
+            given(objectMapper.writeValueAsString(IpfsReadErrorInfoMessage))
+                .willReturn(responseMessage)
+        }
+
+        suppose("asset info IPFS hash will be fetched") {
+            given(assetContractService.getAssetInfoIpfsHash())
+                .willReturn(IpfsHash("testHash").right())
+        }
+
+        suppose("asset info file will be fetched") {
+            given(ipfsRepository.fetchTextFile(IpfsHash("testHash")))
+                .willReturn(IpfsTextFile("{}").right())
+        }
+
+        suppose("asset info JSON is valid") {
+            given(objectMapper.readTree("{}"))
+                .willReturn(ObjectNode(null))
+        }
+
+        val assetCategoryId = AssetCategoryId(BigInteger.valueOf(42L))
+
+        suppose("asset category ID will be fetched") {
+            given(assetContractService.getAssetCategoryId())
+                .willReturn(assetCategoryId.right())
+        }
+
+        suppose("auditing procedure IPFS hash will be fetched") {
+            given(registryContractService.getAuditingProcedureDirectoryIpfsHash(assetCategoryId))
+                .willReturn(IpfsHash("procedureHash").right())
+        }
+
+        suppose("auditing procedure will not be fetched") {
+            given(ipfsRepository.fetchTextFileFromDirectory(IpfsHash("procedureHash"), "audit.js"))
+                .willReturn(IpfsEmptyResponseError(IpfsHash("procedureHash"), "audit.js").left())
+        }
+
+        verify("correct web socket messages are sent") {
+            runBlocking {
+                controller.onOpen(session)
+            }
+
+            then(session)
+                .should(times(1))
+                .sendSync(connectedMessage)
+            then(session)
+                .should(times(1))
+                .sendSync(responseMessage)
+        }
+
+        verify("web socket session was closed") {
+            then(session)
+                .should(times(1))
+                .close(CloseReason.ABNORMAL_CLOSURE)
+        }
+    }
+
+    @Test
+    fun `must successfully start auditing script`() {
```

Comment:
I suppose it is not really needed, but it is good to have some overlapping cases in tests. That way it may be easier to find an error if there is one. This test in particular just checks wether or not `executorService` had a task scheduled at the end of the `onOpen` web socket method. We are not interested here if that task was executed successfully or not - here we only check that something was scheduled. If then scheduling stops working for example, we would have 3 failing tests - and this one would tell us that there is something wrong with scheduling by itself, the other two tell us if there is something wrong after the scheduled task is executed.